lib/eventFactory.js

```diff
@@ -2,9 +2,94 @@
 
 const BaseFactory = require('./baseFactory');
 const Event = require('./event');
+const jobConfigSchema = require('screwdriver-data-schema').config.job;
+const joi = require('joi');
 
 let instance;
 
+/**
+ * Create builds associated with this event
+ * For example, if startFrom ~commit, then create builds with jobs that have requires: ~commit
+ * @method createBuilds
+ * @param  {Pipeline} pipeline                   Pipeline to create builds
+ * @param  {Object}   config
+ * @param  {String}   config.sha                 SHA this project was built on
+ * @param  {String}   config.username            Username of the user that creates this event
+ * @param  {String}   config.scmContext          The scm context to which user belongs
+ * @param  {String}   [config.prRef]             Ref if it's a PR event
+ * @param  {String}   [config.prNum]             PR Number if it's a PR event
+ * @param  {String}   [config.startFrom]         Where the event starts from (jobname or ~commit, ~pr, etc)
+ *                                               Making optional for backward compatibility
+ * @param  {Number}   eventId                    Event's id
+ */
+function createBuilds(pipeline, config, eventId) {
+    const startFrom = config.startFrom;
+
+    // If no startFrom, then do nothing. Remove once we switch to new workflow design
+    if (!startFrom) {
+        return null;
+    }
+
+    /* eslint-disable global-require */
+    const BuildFactory = require('./buildFactory');
```

Comment:
any reason why these are not global ?

Reply:
This is due to a circular dependency issue
https://github.com/screwdriver-cd/models/blob/master/lib/eventFactory.js#L43-L45